### PR TITLE
[Config] [Configuration] Emphasize env vars purpose

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fix https://github.com/symfony/symfony-docs/issues/14745

### CODE REVIEW

File: configuration.rst [hunk 1]
```diff
@@ -599,6 +599,15 @@ The values of these options are resolved at runtime (only once per request, to
 not impact performance) so you can change the application behavior without having
 to clear the cache.
 
+.. caution::
+
+    Environment variables are designed to store information that can
+    dynamically change in a production environment, such as an API
+    key that may have an expiration date. This information could be updated
+    without having to redeploy the application nor clear the cache. Thus, not
```
Comment (94noni) [inline configuration.rst]:
```suggestion
    without having to redeploy the application. Thus, not
```

it is already document in the paragraph above `without having to clear the cache.`

File: configuration.rst [hunk 2]
```diff
@@ -599,6 +599,15 @@ The values of these options are resolved at runtime (only once per request, to
...
+    everything is a good candidate for env vars, and
+    :ref:`parameters <configuration-parameters>` should be used in other cases.
```
Comment (94noni) [inline configuration.rst:609]:
```suggestion
    :ref:`parameters <configuration-parameters>` should be challenged with environment configs.
```

File: configuration.rst [hunk 3]
```diff
@@ -599,6 +599,15 @@ The values of these options are resolved at runtime (only once per request, to
 not impact performance) so you can change the application behavior without having
 to clear the cache.
 
+.. caution::
```
Comment (94noni) [inline configuration.rst]: is it a caution or more like a note ?

**Review by 94noni — COMMENTED** [review]
i like the paragraph mostly
small comments as well

File: configuration.rst (hunk 3 above)
Comment (alexandre-daubois) [reply configuration.rst]: Changed to `note`, you're right

File: configuration.rst (hunk 1 above)
Comment (alexandre-daubois) [reply configuration.rst]: Removed :+1:

File: configuration.rst (hunk 2 above)
Comment (alexandre-daubois) [reply configuration.rst:609]: I don't know, I'm always afraid with the `this should be challenged` wording. I always have the feeling it is very vague and doesn't help to make a decision. Does anyone else have thoughts on this?

Comment (94noni) [reply configuration.rst:609]: oki got your point now

**Review by 94noni — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Alex, this was a nice contribution as always.

However, while merging I notices that the new `note` explained a use case for env vars ... but in the intro of this section we mentioned another use case. So, I removed the `note` and merged its contents into the section intro to make it easier to readers. See https://github.com/symfony/symfony-docs/commit/149876b60ef3ed6c5a3418d0c93538a978b20314

**Pull request merged**